### DISCUSSION TITLE
Some questions about the Jacobian matrix

### DISCUSSION THREAD

**Discussion by HPL001** [idaholab/moose#29524] [Q&A Modules: General]
![image](https://github.com/user-attachments/assets/a43c3c97-7f0d-4909-a848-239d5dbe95ee)
When I was doing the calculations, it reminded me of some caveats about the Jacobian matrix, and I don't know if this is something to be concerned about. If so, how should I solve this problem?

**Comment by GiudGiud** [comment]
Hello 

Jacobian calculation works well for kernels and boundary conditions

but this caveat is for using a User Object to contribute to the Jacobian. This is not very standard. How are you using this feature?

**Reply by GiudGiud** [reply]
The `ComputeGrainForceAndTorque` is the one returning this warning I think.
It should be fine to use. You should just be mindful that if you use more advanced features in the same input, the combination with this experimental feature might not work as expected

**Reply by HPL001** [reply]
Thank you for your reply, it would be great if this warning could not be paid attention to. However, the current input file can be stable for some time, and after a while the linear solver diverges, what is the problem?

**Reply by GiudGiud** [reply]
see this page for more info on troubleshooting failed sovles
https://mooseframework.inl.gov/moose/application_usage/failed_solves.html
